prog.go

```diff
@@ -498,6 +498,23 @@ func (p *Program) Close() error {
 	return p.fd.Close()
 }
 
+// Various options for Run'ing a Program
+type RunOptions struct {
```

Comment:
Name bikeshed. Timo suggested `RunContext`, but then we'd have `RunContext.Context`. `RunOptions` is consistent with other option structs we have in the library, but it's a little bit weird in that we actually write into the `Out` fields. Probably not a big deal, so I'm fine with `RunOptions`.